credentials.go

```diff
@@ -27,62 +27,40 @@ import (
 	"go.ketch.com/lib/orlop/log"
 )
 
-// HasCredentialsConfig denotes that an object provides credentials configuration
-type HasCredentialsConfig interface {
-	GetID() string
-	GetUsername() string
-	GetPassword() string
-}
-
 // CredentialsConfig provides credentials configuration
 type CredentialsConfig struct {
 	ID       string
 	Username string
 	Password string
 }
 
-// GetID returns the ID of the credentials in Vault
-func (c CredentialsConfig) GetID() string {
-	return c.ID
-}
-
-// GetUsername returns a static username
-func (c CredentialsConfig) GetUsername() string {
-	return c.Username
-}
-
-// GetPassword returns a static password
-func (c CredentialsConfig) GetPassword() string {
-	return c.Password
-}
-
 // Credentials provides username/password information
 type Credentials struct {
 	Username string
 	Password string
 }
 
 // GetCredentials retrieves credentials
-func GetCredentials(ctx context.Context, cfg HasCredentialsConfig, vault HasVaultConfig) (*Credentials, error) {
+func GetCredentials(ctx context.Context, cfg CredentialsConfig, vault VaultConfig) (*Credentials, error) {
 	ctx, span := tracer.Start(ctx, "GetCredentials")
 	defer span.End()
 
 	l := log.WithFields(logrus.Fields{
-		"credentials.id": cfg.GetID(),
-		"vault.enabled":  vault != nil && vault.GetEnabled(),
+		"credentials.id": cfg.ID,
+		"vault.enabled":  vault.GetEnabled(),
 	})
 	l.Trace("loading credentials")
 
-	if len(cfg.GetUsername()) > 0 && len(cfg.GetPassword()) > 0 {
+	if len(cfg.Username) > 0 && len(cfg.Password) > 0 {
 		l.Trace("loaded from inline settings")
 
 		return &Credentials{
-			Username: cfg.GetUsername(),
-			Password: cfg.GetPassword(),
+			Username: cfg.Username,
+			Password: cfg.Password,
 		}, nil
 	}
 
-	if len(cfg.GetID()) == 0 || vault == nil || !vault.GetEnabled() {
+	if len(cfg.ID) == 0 || !vault.GetEnabled() {
```

Comment:
Will this panic if vault is null?  We do have a null check previously?

Reply:
No longer possible to be nil.